src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -282,14 +286,14 @@ private static bool IsGetHashCodeInterfaceImplementation(IMethodSymbol method, C
             {
                 // Substitute the type of the first parameter of Equals in the generic interface and then check if that
                 // interface method is implemented by the given method.
-                INamedTypeSymbol iEqualityComparer = WellKnownTypes.GenericIEqualityComparer(compilation);
+                INamedTypeSymbol iEqualityComparer = compilation.GetTypeByMetadataName(typeof(IEqualityComparer<>).FullName);
```

Comment:
Lot of inconsistency in the way well known types are accessed in this file, and now the analyzer also needs to know how to access generic types.